specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/resources.json

```diff
@@ -0,0 +1,6392 @@
+          "description": "The target resource."
+        },
+        "request": {
+          "readOnly": true,
+          "$ref": "#/definitions/HttpMessage",
+          "description": "The HTTP request message."
+        },
+        "response": {
+          "readOnly": true,
+          "$ref": "#/definitions/HttpMessage",
+          "description": "The HTTP response message."
+        }
+      },
+      "description": "Deployment operation properties."
+    },
+    "DeploymentOperation": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Full deployment operation ID."
+        },
+        "operationId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Deployment operation ID."
+        },
+        "properties": {
+          "$ref": "#/definitions/DeploymentOperationProperties",
+          "description": "Deployment properties."
+        }
+      },
+      "description": "Deployment operation information."
+    },
+    "DeploymentOperationsListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DeploymentOperation"
+          },
+          "description": "An array of deployment operations."
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URL to use for getting the next set of results."
+        }
+      },
+      "description": "List of deployment operations."
+    },
+    "ResourceProviderOperationDisplayProperties": {
+      "properties": {
+        "publisher": {
+          "type": "string",
+          "description": "Operation description."
+        },
+        "provider": {
+          "type": "string",
+          "description": "Operation provider."
+        },
+        "resource": {
+          "type": "string",
+          "description": "Operation resource."
+        },
+        "operation": {
+          "type": "string",
+          "description": "Resource provider operation."
+        },
+        "description": {
+          "type": "string",
+          "description": "Operation description."
+        }
+      },
+      "description": "Resource provider operation's display properties."
+    },
+    "Resource": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID"
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "extendedLocation": {
+          "$ref": "#/definitions/ExtendedLocation",
+          "description": "Resource extended location."
```

Comment:
since this is new, we should have the correct x-ms-mutability (location should as well, but this would technically be a breaking change).  If this is determined by the service, it should be set as readOnly.  If it can be set on creation only, it should be "read", "create".  If it is settable on create and update, then you can leave it as is.

Reply:
udpated.